### PR TITLE
MP Conc TCK execution - use Arq. extension to add implementation clas…

### PR DESCRIPTION
**Pull request by manovotn**
…ses.

Bump CDI version to 2.0 and Weld to 3.x

These changes should solve the producer problem (TCK test passes for me after extension removal) as well as update CDI and Weld versions.
This will only work if test archive is WAR, if there are to be any JARs, we will need more logic (see the `TODO`)

### CODE REVIEW

**Comment by manovotn** [comment]
CI build failure is expected - TCKs are just a PR on spec side, hence the artifact is non-existent until you build it yourself ;-)

**Comment by FroMage** [comment]
That's great! But doesn't it also mean the spec's TCK needs a CDI version bump?

**Comment by manovotn** [comment]
Spec TCK runs on CDI 1.2?
I must have missed that. In that case yea, it needs bump as well.

File: tck/src/test/java/io/smallrye/concurrency/tck/SmallRyeConcurrencyArchiveProcessor.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+        if (applicationArchive instanceof WebArchive) {
+            WebArchive archive = (WebArchive) applicationArchive;
+            // we need to create new archives and add them as dependencies, they have to be instance of ArchiveAsset in order to pass
+            // this check https://github.com/arquillian/arquillian-container-weld/blob/2.0.0.Final/impl/src/main/java/org/jboss/arquillian/container/weld/embedded/Utils.java#L73
+            for (int i = 0; i < dependencies.length; i++) {
+                JavaArchive newJar = ShrinkWrap.createFromZipFile(JavaArchive.class, dependencies[0]);
+                archive.addAsLibrary(newJar);
+            }
+        }
+        if (applicationArchive instanceof WebArchive) {
```
[the remark this answers is not in the document]
Comment (manovotn) [reply tck/src/test/java/io/smallrye/concurrency/tck/SmallRyeConcurrencyArchiveProcessor.java]: Well, yea, that's typo indeed. Thanks.
But I don't even know if we need to solve that case yet, depending on how the MP Conc TCK will look like.

[the remark this answers is not in the document]
Comment (Ladicek) [reply tck/src/test/java/io/smallrye/concurrency/tck/SmallRyeConcurrencyArchiveProcessor.java]: Sure, and I'm sorry I can't help with that! The TODO in the comment just didn't match with the code :-)

**Comment by FroMage** [comment]
Thanks a lot :)

**Comment by FroMage** [comment]
Merged, since MP-Concurrency PR on TCK was merged too.

File: tck/src/test/java/io/smallrye/concurrency/tck/SmallRyeConcurrencyArchiveProcessor.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        // SmallRye Conc. CDI impl
+        String dependencyToAdd = "io.smallrye:smallrye-concurrency-cdi";
+        File[] dependencies = Maven.resolver().loadPomFromFile(new File("pom.xml")).resolve(dependencyToAdd).withTransitivity().asFile();
+        // in case to WAR, we add it as a library, as would user in their app
+        if (applicationArchive instanceof WebArchive) {
+            WebArchive archive = (WebArchive) applicationArchive;
+            // we need to create new archives and add them as dependencies, they have to be instance of ArchiveAsset in order to pass
+            // this check https://github.com/arquillian/arquillian-container-weld/blob/2.0.0.Final/impl/src/main/java/org/jboss/arquillian/container/weld/embedded/Utils.java#L73
+            for (int i = 0; i < dependencies.length; i++) {
+                JavaArchive newJar = ShrinkWrap.createFromZipFile(JavaArchive.class, dependencies[0]);
```
Comment (FroMage) [inline tck/src/test/java/io/smallrye/concurrency/tck/SmallRyeConcurrencyArchiveProcessor.java]: @manovotn should this have been `dependencies[i]`?

**Pull request merged**